Source/WebCore/platform/graphics/TrackBuffer.cpp

```diff
@@ -100,22 +100,35 @@ bool TrackBuffer::updateMinimumUpcomingPresentationTime()
     return true;
 }
 
-bool TrackBuffer::reenqueueMediaForTime(const MediaTime& time, const MediaTime& timeFudgeFactor)
+bool TrackBuffer::reenqueueMediaForTime(const MediaTime& time, const MediaTime& timeFudgeFactor, bool isEnded)
 {
     m_decodeQueue.clear();
 
     m_highestEnqueuedPresentationTime = MediaTime::invalidTime();
     m_lastEnqueuedDecodeKey = { MediaTime::invalidTime(), MediaTime::invalidTime() };
     m_enqueueDiscontinuityBoundary = time + m_discontinuityTolerance;
 
+    if (m_samples.empty())
+        return false;
+
     // Find the sample which contains the current presentation time.
     auto currentSamplePTSIterator = m_samples.presentationOrder().findSampleContainingPresentationTime(time);
 
-    if (currentSamplePTSIterator == m_samples.presentationOrder().end())
-        currentSamplePTSIterator = m_samples.presentationOrder().findSampleStartingOnOrAfterPresentationTime(time);
+    // Find the next sample, so long as its presentation start time is within the timeFudgeFactor.
+    if (currentSamplePTSIterator == m_samples.presentationOrder().end()) {
+        auto nextSampleIterator = m_samples.presentationOrder().findSampleStartingOnOrAfterPresentationTime(time);
+        if ((nextSampleIterator->first - time) <= timeFudgeFactor)
+            currentSamplePTSIterator = nextSampleIterator;
```

Comment:
Is this needed here?

For the issue at and, below is all necessary right?

Reply:
This is just refactoring the existing check.